Adjust Keda docs according to the documentation requirements (#156)
kyma-project/keda-manager: docs/contributor/02-01-installation.md

# Install Keda Manager 

- [Install Keda Manager](#install-keda-manager)
  - [Prerequisites](#prerequisites)
  - [Install Keda Manager without Kyma](#install-keda-manager-without-kyma)
  - [Build and publish images manually](#build-and-publish-images-manually)
  - [Install Keda module manually](#install-keda-module-manually)
  - [Install on Kyma runtime](#install-on-kyma-runtime)
    - [Lifecycle management of Keda Manager in Kyma](#lifecycle-management-of-keda-manager-in-kyma)
  - [Run Keda module locally with Kyma and Lifecycle Manager on k3d](#run-keda-module-locally-with-kyma-and-lifecycle-manager-on-k3d)
    - [Run Keda module locally on bare k3d](#run-keda-module-locally-on-bare-k3d)
    - [Install Keda module on remote Kyma runtime](#install-keda-module-on-remote-kyma-runtime)

Learn how to install Keda Manager locally (on k3d) or on your remote cluster.

## Prerequisites

- Access to a Kubernetes (v1.24 or higher) cluster or [k3d](https://k3d.io/v5.4.6/)
- [Go](https://go.dev/)
- [Docker](https://www.docker.com/)
- [kubectl](https://kubernetes.io/docs/tasks/tools/)
- [kubebuilder](https://book.kubebuilder.io/)

> ## Install Keda Manager without Kyma 

Reviewer: ```suggestion
## Install Keda Manager from the local sources 
```

For the day-to-day development on your machine, you don't always need to have it controlled by Kyma's Lifecycle Manager.

Run the following commands to deploy Keda Manager on a target Kubernetes cluster (for example, on k3d):

1. Clone the project.

   ```bash
   git clone https://github.com/kyma-project/keda-manager.git && cd keda-manager/
   ```

2. Set the Keda Manager image name.

   > NOTE: You can use local k3d registry or your Docker Hub account to push intermediate images.  
   ```bash
   export IMG=<DOCKER_USERNAME>/custom-keda-manager:0.0.2
   ```

3. Verify the compatibility.

   ```bash
   make test
   ```
4. Build and push the image to the registry.

   ```bash
   make module-image
   ```
5. Create a target Namespace.

   ```bash
   kubectl create ns kyma-system
   ```

6. Deploy Keda Manager.

   ```bash
   make deploy
   ```

7. Verify if Keda Manager is deployed.

   ```bash
   kubectl get deployments -n kyma-system
   ```

   You should get a result similar to this example:

   ```
   NAME                             READY   UP-TO-DATE   AVAILABLE   AGE
   keda-manager            1/1     1            1           1m
   ```

> ## Build and publish images manually

Reviewer: Lets remove this section ("Build and publish images manually") - its confusing

1. Export the required environmental variables

   ```
   export IMG="IMG"           // Keda Manager's image
   export REGISTRY="REGISTRY" // the OCI registry the module will be published to
   ```

2. Run the following recipe to build and publish module

   ```
   make module-build \
     IMG=${IMG} \
     REGISTRY={REGISTRY}
   ```

> ## Install Keda module manually

Reviewer: ```suggestion
## Install Keda module from the latest release
```

1. Apply the following script to install Keda Manager:

   ```bash
   kubectl create ns kyma-system
   kubectl apply -f https://github.com/kyma-project/keda-manager/releases/latest/download/keda-manager.yaml
   ```

2. To get KEDA installed, apply the sample Keda CR:

   ```bash
   kubectl apply -f config/samples/operator_v1alpha1_keda_k3d.yaml
   ```

> ## Install on Kyma runtime

Reviewer: I believe that this section is neiither relevant for OS users nor for SKR users. Morover, for SKR users it may confuse them, as they will use the internal documentation for how to enable modules on kyma runtimes with lifecycle manager.

Lets delete this section ("Install on Kyma runtime")

> When you enable the Keda module using your Kyma runtime Kyma custom resource (CR), the Lifecycle Manager downloads the bundled package of the Keda Manager and installs it. Additionally, it applies a sample Keda CR, which triggers Keda Manager to install the Keda module.

Reviewer: Lets move this subsection ("Lifecycle management of Keda Manager in Kyma") to a separate file in the contributor folder. Lets keep the description, diagram, but please drop the command to enable module via kyma CLI command

> ![a](/docs/assets/keda-lm-overview.drawio.svg)

Reviewer 2: The diagram is not formatted according to [our guidelines](https://github.com/kyma-project/community/blob/main/docs/guidelines/content-guidelines/09-diagrams.md).

To enable the Keda module run:

   ```bash
   kyma alpha enable module keda -c fast
   ```

> ## Run Keda module locally with Kyma and Lifecycle Manager on k3d

Reviewer: ```suggestion
## Make targets to run Keda module locally on k3d
### Run Keda module with Lifecycle Manager
```

Use the dedicated `make` target (in the `hack` folder).

   ```bash
   make -C hack/local run-with-lifecycle-manager
   ```
   

> ### Run Keda module locally on bare k3d

Reviewer: ```suggestion
### Run Keda module on bare k3d
```
